Habitica/src/main/java/com/habitrpg/android/habitica/ui/adapter/social/challenges/ChallengeTasksRecyclerViewAdapter.java

```diff
@@ -0,0 +1,194 @@
+package com.habitrpg.android.habitica.ui.adapter.social.challenges;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.github.underscore.$;
+import com.habitrpg.android.habitica.R;
+import com.habitrpg.android.habitica.components.AppComponent;
+import com.habitrpg.android.habitica.helpers.TaskFilterHelper;
+import com.habitrpg.android.habitica.models.tasks.Task;
+import com.habitrpg.android.habitica.ui.adapter.tasks.SortableTasksRecyclerViewAdapter;
+import com.habitrpg.android.habitica.ui.viewHolders.tasks.BaseTaskViewHolder;
+import com.habitrpg.android.habitica.ui.viewHolders.tasks.DailyViewHolder;
+import com.habitrpg.android.habitica.ui.viewHolders.tasks.HabitViewHolder;
+import com.habitrpg.android.habitica.ui.viewHolders.tasks.RewardViewHolder;
+import com.habitrpg.android.habitica.ui.viewHolders.tasks.TodoViewHolder;
+
+import java.util.List;
+
+import rx.functions.Action1;
+
+public class ChallengeTasksRecyclerViewAdapter
+        extends SortableTasksRecyclerViewAdapter<BaseTaskViewHolder> {
+    public static final String TASK_TYPE_ADD_ITEM = "ADD_ITEM";
+
+    private static final int TYPE_HEADER = 0;
+    private static final int TYPE_HABIT = 1;
+    private static final int TYPE_DAILY = 2;
+    private static final int TYPE_TODO = 3;
+    private static final int TYPE_REWARD = 4;
+    private static final int TYPE_ADD_ITEM = 5;
+
+    private int dailyResetOffset = 0;
+    private Action1<Task> addItemCallback;
```

Comment:
Please use a PublishSubject here. That's what we also started to use in other placed for this kind of thing.